include/nigiri/loader/gtfs/shape.h

```diff
@@ -11,8 +16,21 @@ struct shape_state {
   std::size_t last_seq_{};
 };
 
-using shape_id_map_t = hash_map<std::string, shape_state>;
+struct shape_loader_state {
+  hash_map<std::string, shape_state> id_map_{};
+  // Stores median for each leg to allow small errors for each stop
+  vecvec<shape_idx_t, double> distance_edges_{};
+  shape_idx_t index_offset_;
+};
+
+shape_loader_state parse_shapes(std::string_view const, shapes_storage&);
 
-shape_id_map_t parse_shapes(std::string_view const, shapes_storage_t&);
+template <typename DoubleRange>
+  requires std::ranges::range<DoubleRange> &&
+           std::is_same_v<std::ranges::range_value_t<DoubleRange>, double>
+inline bool valid_distances(DoubleRange distances) {
```

Comment:
```suggestion
bool valid_distances(DoubleRange distances) {
```

Templates do not need `inline` and usually the compiler does a better job at deciding what should be really inlined. So `inline` is mostly a instruction to the linker, which is not needed in this case (as it's a template).

Reply:
Thank you for the reminder.
Due to some changes, the function has been removed entirely.